File: scheduler/core/src/main/java/io/orkes/conductor/scheduler/model/NextScheduleResult.java
package io.orkes.conductor.scheduler.model;

import java.time.ZonedDateTime;

import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@AllArgsConstructor
public class NextScheduleResult {

    private final ZonedDateTime nextRunTime;

    private final String zoneId;

    public static NextScheduleResult of(ZonedDateTime nextRunTime, String zoneId) {
        return new NextScheduleResult(nextRunTime, zoneId);
    }

    public boolean hasNextRunTime() {
        return nextRunTime != null;
    }
}

File: scheduler/core/src/main/typescript/io/orkes/conductor/scheduler/model/NextScheduleResult.ts
export class NextScheduleResult {
    constructor(
        public readonly nextRunTime: Date | null,
        public readonly zoneId: string
    ) {}

    static of(nextRunTime: Date | null, zoneId: string): NextScheduleResult {
        return new NextScheduleResult(nextRunTime, zoneId);
    }

    hasNextRunTime(): boolean {
        return this.nextRunTime != null;
    }
}
